src/hacbs/components/ApplicationDetails/tabs/overview/visualization/hooks/useCommitWorkflowData.ts

```diff
@@ -0,0 +1,314 @@
+import { RunStatus } from '@patternfly/react-topology';
+import { pipelineRunStatus } from '../../../../../../../shared';
+import { ComponentKind } from '../../../../../../../types';
+import { useNamespace } from '../../../../../../../utils/namespace-context-utils';
+import { PipelineRunLabel } from '../../../../../../consts/pipelinerun';
+import {
+  useComponents,
+  useEnvironments,
+  useIntegrationTestScenarios,
+  useReleasePlans,
+  useBuildPipelines,
+  useReleases,
+  useTestPipelines,
+  useApplicationSnapshotsEB,
+} from '../../../../../../hooks';
+import { Commit, PipelineRunKind } from '../../../../../../types';
+import {
+  EnvironmentKind,
+  IntegrationTestScenarioKind,
+  ReleaseKind,
+  ReleasePlanKind,
+  SnapshotEnvironmentBinding,
+} from '../../../../../../types/coreBuildService';
+import { CommitComponentResource, Workflow, WorkflowNode, WorkflowNodeType } from '../types';
+import {
+  getLastEnvironments,
+  getLatestResource,
+  workflowToNodes,
+} from '../utils/visualization-utils';
+
+export const useCommitWorkflowData = (commit: Commit): [nodes: WorkflowNode[], loaded: boolean] => {
+  const namespace = useNamespace();
+  const applicationName = commit.application;
+  const [components, componentsLoaded] = useComponents(namespace, applicationName);
+  const [integrationTests, integrationTestsLoaded] = useIntegrationTestScenarios(
+    namespace,
+    applicationName,
+  );
+  const [environments, environmentsLoaded] = useEnvironments(namespace);
+  const [releasePlans, releasePlansLoaded] = useReleasePlans(namespace);
+  const [releases, releasesLoaded] = useReleases(namespace);
+  const [buildPipelines, buildPipelinesLoaded] = useBuildPipelines(
+    namespace,
+    applicationName,
+    commit.sha,
+  );
+  const [testPipelines, testPipelinesLoaded] = useTestPipelines(
+    namespace,
+    applicationName,
+    commit.sha,
+  );
+
+  const [applicationSnapshotsEB, applicationSnapshotsLoaded] = useApplicationSnapshotsEB(
+    namespace,
+    applicationName,
+  );
+
+  const allResourcesLoaded: boolean =
+    componentsLoaded &&
+    integrationTestsLoaded &&
+    buildPipelinesLoaded &&
+    testPipelinesLoaded &&
+    environmentsLoaded &&
+    applicationSnapshotsLoaded &&
+    releasesLoaded &&
+    releasePlansLoaded;
+
+  if (!allResourcesLoaded) {
+    return [[], allResourcesLoaded];
+  }
+
+  const commitComponents = buildPipelines.map(
+    (bp) => bp.metadata.labels[PipelineRunLabel.COMMIT_COMPONENT_LABEL],
+  );
+
+  const commitWorkflowData: { [key: string]: any } = components.reduce(
```

Comment:
There's opportunity to memoize some values here such that we don't need to recompute everything if only a single value changes.

Reply:
That being said, it's something that can be improved on in future PRs and not needed right away.